### PR TITLE
Command to to calculate the total of all coinbase transactions so far #1196

### PR DESCRIPTION
**Pull request by revler1082**
Added a print_coinbase_tx_sum option

### CODE REVIEW

**Comment by luigi1111** [comment]
I think this is neat, but I'm not sure how useful it is without the ability to distinguish between emission and fees...

File: src/cryptonote_core/cryptonote_core.cpp [hunk 1]
```diff
@@ -616,6 +616,34 @@ namespace cryptonote
...
+      coinbase_amount = get_outs_money_amount(b.miner_tx);
+      this->get_transactions(b.tx_hashes, txs, missed_txs);      
+      BOOST_FOREACH(const auto& tx, txs)
+      {
+        tx_fee_amount += get_tx_fee(tx);
+      }
+      
+      emission_amount += coinbase_amount - tx_fee_amount;
+      total_fee_amount += tx_fee_amount;
+      coinbase_amount = 0;
```
Comment (moneromooo-monero) [inline src/cryptonote_core/cryptonote_core.cpp]: why ?

File: src/cryptonote_core/cryptonote_core.h [hunk 2]
```diff
@@ -600,6 +600,13 @@ namespace cryptonote
       */
      size_t get_block_sync_size() const { return block_sync_size; }
 
+     /**
+      * @brief get the sum of coinbase tx amounts between blocks
+      *
+      * @return the number of blocks to sync in one go
+      */
+     std::pair<uint64_t, uint64_t> get_coinbase_tx_sum(const uint64_t start_offset, const uint64_t count);
```
Comment (moneromooo-monero) [inline src/cryptonote_core/cryptonote_core.h]: Should be const

File: src/daemon/command_parser_executor.cpp [hunk 3]
```diff
@@ -452,5 +452,27 @@ bool t_command_parser_executor::output_histogram(const std::vector<std::string>&
   return m_executor.output_histogram(min_count, max_count);
 }
 
+bool t_command_parser_executor::print_coinbase_tx_sum(const std::vector<std::string>& args)
+{
+  if(!args.size())
+  {
+    std::cout << "need block height parameter" << std::endl;
+    return false;
+  }
```
Comment (moneromooo-monero) [inline src/daemon/command_parser_executor.cpp]: Maybe should default to use current blockchain height, though that's a question of taste I guess

File: src/cryptonote_core/cryptonote_core.cpp [hunk 4]
```diff
@@ -616,6 +616,34 @@ namespace cryptonote
...
+  std::pair<uint64_t, uint64_t> core::get_coinbase_tx_sum(const uint64_t start_offset, const uint64_t count)
+  {
+    std::list<block> blocks;
+    std::list<transaction> txs;
+    std::list<crypto::hash> missed_txs;
+    uint64_t coinbase_amount = 0;
+    uint64_t emission_amount = 0;
+    uint64_t total_fee_amount = 0;
+    uint64_t tx_fee_amount = 0;
+    this->get_blocks(start_offset, count, blocks);
```
Comment (moneromooo-monero) [inline src/cryptonote_core/cryptonote_core.cpp]: This is also likely to take a LOT of memory. The database keeps track of the coins emitted up to a given block. Subtracting those on the end/start blocks will be a lot faster and less memory hungry.

**Comment by moneromooo-monero** [comment]
Mostly fine, a few comments left on the diff

**Comment by schnerchi** [comment]
This PR breaks compiling on macos sierra.

File: src/cryptonote_core/cryptonote_core.cpp (hunk 4 above)
Comment (revler1082) [inline src/cryptonote_core/cryptonote_core.cpp]: Looking at the already_generated_coins function, does this return only the emission portion? I'm also moving the calculations to core_rpc_server in order to not pollute cryptonote_core, and looping over the blocks one at a time to get tx fees. 

You think I should just minimize the scope of the function to just show newly minted coins, which should be pretty quick?

Thanks again for your time/help.

Comment (moneromooo-monero) [inline src/cryptonote_core/cryptonote_core.cpp]: Yes, already_generated_coins does the emission portion. If you want the fees, you're right you'd have to go through the entire blockchain to scan.

**Pull request merged**